applications/forms.py

```diff
@@ -137,6 +139,18 @@ def clean_other_gender(self):
             raise forms.ValidationError("Please enter this field or select 'Prefer not to answer'")
         return data
 
+    def clean_origin(self):
+        origin = self.cleaned_data['origin']
+        response = requests.get('https://api.teleport.org/api/cities/', params={'search': origin})
+        data = response.json()['_embedded']['city:search-results']
+        if origin == "Others":
+            origin_verified = origin
```

Comment:
Can we move this up to avoid having to call the API if user has set origin as "Others"?  No need to do a request if we already know that origin=='Others'